sdk/identity/azure-identity/tests/test_vscode_credential_async.py

```diff
@@ -95,9 +96,13 @@ async def test_cache_refresh_token():
 
 @pytest.mark.asyncio
 async def test_no_obtain_token_if_cached():
+    def mock_should_refresh(token):
+        return False
+
     expected_token = AccessToken("token", 42)
 
     mock_client = mock.Mock(spec=object)
+    mock_client.should_refresh = mock_should_refresh
```

Comment:
Just FYI, the `Mock` API provides an easier way to do this:
```suggestion
    expected_token = AccessToken("token", 42)

    mock_client = mock.Mock(should_refresh=lambda: False)
```